### PR TITLE
Fix GCC12 warnings in fidlib.c 

### PR DESCRIPTION
**Pull request by daschuer**
This should fix the warnings reported here: #11409 
This also includes a bit of refactoring, because the indent of the affected function was broken. 
Please see single commits for review.

### CODE REVIEW

**Review by Swiftb0y — APPROVED** [review]
LGTM, waiting for CI. Would be nice to drop fidlib entirely in the future. Doesn't seem to have aged very well judging by the amount of warnings.

**Comment by uklotzde** [comment]
Reformattings should be done in a separate PR without changing any code at all. I guess no one understands what this legacy, hardcore C code really does. Not reviewable.

**Comment by daschuer** [comment]
@uklotzde That's why I have ask to review the single commits.

**Comment by Swiftb0y** [comment]
While I don't understand the mathematics behind the code, I was able to go through the changes carefully and confirm that the behavior is still the same. It's just written differently so the compiler can reason about it better so it doesn't result in false-positive warnings.

File: lib/fidlib/fidlib.c [hunk 1]
```diff
@@ -2334,13 +2334,16 @@ fid_parse(double rate, char **pp, FidFilter **ffp) {
...
+      size_t curr_offset= (char*)curr - rv;         \
+      tmp= (char*)realloc(rv, new_size);            \
       if (!tmp) {                                   \
          error("Out of memory");                    \
-      }                                             \
-      rvend= (rvend - rv) * 2 + tmp;                \
-      rvp= (rvp - rv) + tmp;                        \
-      curr= (FidFilter*)(((char*)curr) - rv + tmp); \
+      }                                            \
+      rvend= tmp + new_size;;                       \
```
Comment (Swiftb0y) [inline lib/fidlib/fidlib.c:2344]:
```suggestion
      }                                             \
      rvend= tmp + new_size;                        \
```

File: lib/fidlib/fidlib.c [hunk 2]
```diff
@@ -2359,19 +2362,22 @@ fid_parse(double rate, char **pp, FidFilter **ffp) {
          case ';':
          case ')':
          case ']':
-         case '}':
+         case '}': {
             // End of filter, return it
-            tmp= (char*)realloc(rv, (rvp-rv) + xtra);
+            size_t new_size= (rvp-rv) + xtra;
+            size_t rvp_offset= rvp - rv;
```
Comment (uklotzde) [inline lib/fidlib/fidlib.c:2368]: Calculating `rvp_offset` effectively twice obfuscates the code. Please reorder.

**Comment by daschuer** [comment]
Done

**Review by Swiftb0y — COMMENTED** [review]
last commit messed up indentation. feel free to amend and force push

File: lib/fidlib/fidlib.c [hunk 3]
```diff
@@ -2364,8 +2364,8 @@ fid_parse(double rate, char **pp, FidFilter **ffp) {
          case ']':
          case '}': {
             // End of filter, return it
-            size_t new_size= (rvp-rv) + xtra;
-            size_t rvp_offset= rvp - rv;
+	    size_t rvp_offset= rvp - rv;
+            size_t new_size= rvp_offset + xtra;
```
[the remark this answers is not in the document]
Comment (Swiftb0y) [reply lib/fidlib/fidlib.c:2368]: Ah, I see, the problem is from mixing tabs and spaces...

**Comment by daschuer** [comment]
Done

**Pull request merged**